mbed_devices/_internal/windows/windows_identifier.py

```diff
@@ -0,0 +1,68 @@
+"""Defines a Windows identifier."""
+from typing import NamedTuple, Any, Optional
+
+from mbed_devices._internal.windows.component_descriptor_utils import is_undefined_value
+
+INSTANCE_ID_CHARACTER = "&"
+
+
+class WindowsUID(NamedTuple):
+    """Definition of a Windows Universal ID."""
+
+    UID: str  # Processed string which may be the closest to the Universal ID.
+    raw_uid: str  # The String from which the UID was derived. Raw form as it can be a
+    # combination of other elements e.g. 8&3432E179&0&000
+    serial_number: Optional[str]  # a serial number
```

Comment:
What's `serial_number` and how does it differ from `UID`? We seem to check they are equal to each other in the `__eq__` method.

Reply:
tried to comment more

Reply:
e3b58499695e9796bf1b0ed63d97dd330c1692a8